build-executor/src/main/java/org/jboss/pnc/executor/DefaultBuildExecutor.java

```diff
@@ -128,6 +131,58 @@ public BuildExecutionSession startBuilding(
         return buildExecutionSession;
     }
 
+    private CompletedBuild optionallyEnableSsh(BuildExecutionSession session, CompletedBuild completedBuild) {
+        DebugData debugData = session.getDebugData();
+        if (debugData.isDebugEnabled()) {
+            debugData.getSshServiceInitializer().accept(debugData);
+            String text = "\n" +
+                    "-------------------------------------------------------\n" +
+                    "SSH server enabled.\n " +
+                    "Log in using password: " + debugData.getPassword() + " and command: \n" +
+                    "ssh root@" + debugData.getSshHost() + " -p " + debugData.getSshPort() + "\n" +
+                    "-------------------------------------------------------\n";
+            return appendToBuildLog(completedBuild, text);
+        }
+        return completedBuild;
+    }
+
+    /**
+     * Ugly hack to display ssh access data in the build log.
+     * Ideally should be replaced with a new widget on build result page
+     *
+     * @param completedBuild build data
+     * @param text message to append to the build log
+     * @return build data with the message appended to the log
+     */
+    private CompletedBuild appendToBuildLog(final CompletedBuild completedBuild, final String text) {
+        return new CompletedBuild() {
+
+            private BuildDriverResult decorateWithSshAccessData(BuildDriverResult result) {
+                return new BuildDriverResult() {
+                    @Override
+                    public String getBuildLog() {
+                        return result.getBuildLog() + text;
```

Comment:
I find this part particuralry ugly.

@matejonnet do you think it's acceptable?